From: SeongJae Park <sj@kernel.org>
Date: Tue, 2 Dec 2025 22:32:56 -0800
Subject: [PATCH] mm/damon/core: report access-generated thread id of the fault
 event

Page fault event reporting function can know which thread is making the
access.  Add the information to the report.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c9d2383ac2bee..a219b822891a4 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1789,6 +1789,7 @@ void damon_report_page_fault(struct vm_fault *vmf, bool huge_pmd)
 	struct damon_access_report access_report = {
 		.size = 1,	/* todo: set appripriately */
 		.cpu = smp_processor_id(),
+		.tid = task_pid_vnr(current),
 	};
 
 	if (huge_pmd)
-- 
2.47.3

